opacus/utils/uniform_sampler.py

```diff
@@ -0,0 +1,34 @@
+import torch
+from torch.utils.data import Sampler
+
+
+class UniformWithReplacementSampler(Sampler):
```

Comment:
How is this different from the standard `RandomSampler` in PyTorch? https://pytorch.org/docs/stable/data.html#torch.utils.data.RandomSampler

It can accept `replacement=True` (though normally people use it with that set to False)

Reply:
I did not succeed to implement the batch construction from Mironov et al [1] with RandomSampler. With this convention, each batch is sampled according to the Sampled Gaussian Mechanism SGM. That means that to build one batch, each sample in the dataset is selected with probability q (called sample rate). Thus, the size of each batch is not fixed. In addition, each batch is built independently from the other batches.

If you are able to implement batch construction according to SGM with RandomSampler, I am very interesting to have access to this implementation.

[1] Rényi Differential Privacy of the Sampled Gaussian Mechanism